### PR TITLE
Update en APIs of LoDTensor

### PR DESCRIPTION
**Pull request by zhiqiu**
This PR updates en document of LoDTensor's APIs.

### CODE REVIEW

**Comment by zhiqiu** [comment]
Corresponding Chinese doc, [PR1212](https://github.com/PaddlePaddle/FluidDoc/pull/1212).
image

File: paddle/fluid/pybind/pybind.cc [hunk 1]
```diff
@@ -1121,6 +1231,12 @@ All parameter, weight, gradient are variables in Paddle.
            },
            py::arg("tensor"), R"DOC(
              Append a LoDensor to LoDTensorArray.
+              
+             Args:
+                   tensor (LoDTensor): The LoDTensor to be appended.
+
+             Returns:
```
Comment (sneaxiy) [inline paddle/fluid/pybind/pybind.cc:1251]: Void -> None.

File: paddle/fluid/pybind/pybind.cc [hunk 2]
```diff
@@ -476,10 +582,14 @@ PYBIND11_MODULE(core_noavx, m) {
...
-           Return the sequence length of the LoDTensor corresponding to LoD.
+           Return the recursive sequence lengths corresponding to of the LodD 
+           of the LoDTensor.
 
            Returns:
-               out (List[List[int]): the sequence lengths.
+               The recursive sequence lengths.
+
+           Return Type:
+               List[List[int].
```
Comment (sneaxiy) [inline paddle/fluid/pybind/pybind.cc]: `List[List[int]` -> `List[List[int]]`.

File: paddle/fluid/pybind/pybind.cc [hunk 3]
```diff
@@ -303,39 +364,82 @@ PYBIND11_MODULE(core_noavx, m) {
...
+    is 2 (e.g., number of samples is 2), the length of second logical sequence 
+    is 3, and the total length is 5. The data of the first logical sequence is 
+    [1, 2], [3, 4], and the data of the second logical sequence is [5, 6], 
+    [7, 8], [9, 10]. The data dimension of each sample is 2. So, the final 
+    shape of the LoDTensor is [5, 2], of which 5 is the total length and 2 is 
+    the dimension of each sample.
+    
+    Logically, we can represent the variable-length sequence in two ways: one 
+    is in the form of recursive sequence lengths, that is, 
+    x.recursive_sequence_lengths=[[2, 3]; the other is in the form of offsets,
```
Comment (sneaxiy) [inline paddle/fluid/pybind/pybind.cc]: `[[2, 3]` -> ` [[2, 3]]`.

File: paddle/fluid/pybind/pybind.cc [hunk 4]
```diff
@@ -303,39 +364,82 @@ PYBIND11_MODULE(core_noavx, m) {
...
+    is 3, and the total length is 5. The data of the first logical sequence is 
+    [1, 2], [3, 4], and the data of the second logical sequence is [5, 6], 
+    [7, 8], [9, 10]. The data dimension of each sample is 2. So, the final 
+    shape of the LoDTensor is [5, 2], of which 5 is the total length and 2 is 
+    the dimension of each sample.
+    
+    Logically, we can represent the variable-length sequence in two ways: one 
+    is in the form of recursive sequence lengths, that is, 
+    x.recursive_sequence_lengths=[[2, 3]; the other is in the form of offsets, 
+    that is, x.lod=[[0, 2, 2+3]. These two representations are equivalent, and
```
Comment (sneaxiy) [inline paddle/fluid/pybind/pybind.cc]: `[[0, 2, 2+3]` -> `[[0, 2, 2+3]]`

File: paddle/fluid/pybind/pybind.cc [hunk 5]
```diff
@@ -303,39 +364,82 @@ PYBIND11_MODULE(core_noavx, m) {
...
+    From level = 0, there are two logical sequences, the length of which is 
+    2 and 1, respectively, indicating that the first logical sequence contains 
+    two sub-sequences and the second logical sequence contains one sub-sequence. 
+    From level = 1, the lengths of two sub-sequences contained by the first 
+    logical sequence is 2 and 2, and the length of sub-sequence contained by 
+    the second logical sequence is 3.
+      
+    Therefore, the LoDTensor is represented in the form of recursive sequence 
+    lengths as y.recursive_sequence_lengths=[[2,1], [2,2,3]]; and equally, in 
+    the form of offset, it is represented as y.lod=[[0,2,3], [0,2,4,7].
```
Comment (sneaxiy) [inline paddle/fluid/pybind/pybind.cc]: `[[0,2,3], [0,2,4,7]` -> `[[0,2,3], [0,2,4,7]]`

**Review by sneaxiy — COMMENTED** [review]
Try to use the newest preview tools to show whether the style of `Return type` is right.

File: paddle/fluid/pybind/pybind.cc (hunk 1 above)
Comment (zhiqiu) [reply paddle/fluid/pybind/pybind.cc:1251]: Done, thanks.

File: paddle/fluid/pybind/pybind.cc (hunk 2 above)
Comment (zhiqiu) [reply paddle/fluid/pybind/pybind.cc]: Done, thanks.

File: paddle/fluid/pybind/pybind.cc (hunk 3 above)
Comment (zhiqiu) [reply paddle/fluid/pybind/pybind.cc]: Done, thanks.

File: paddle/fluid/pybind/pybind.cc (hunk 5 above)
Comment (zhiqiu) [reply paddle/fluid/pybind/pybind.cc]: Done, thanks.

File: paddle/fluid/pybind/pybind.cc (hunk 4 above)
Comment (zhiqiu) [reply paddle/fluid/pybind/pybind.cc]: Done, thanks.

**Comment by zhiqiu** [comment]
Follow comments.
image

**Review by sneaxiy — APPROVED** [review]
LGTM.

**Review by lanxianghit — APPROVED** [review]

**Review by wanghaoshuang — APPROVED** [review]

**Pull request merged**